### PR TITLE
speak: add "discard ongoing recording" shortcut

### PR DESCRIPTION
**Pull request by olejorgenb**
Allow the user to signal that he messed up and want to begin again.

Could also consider starting recording again immediately? (this would be my personal preference, but without extra work it isn't really visible to the user that something happened. If there was a duration counter visible it would be less of an issue.)

Related to: #1052, #2105

### CODE REVIEW

File: web/locales/en/messages.ftl [hunk 1]
```diff
@@ -959,4 +962,4 @@ contribution-just-unsure-title = Just Unsure?
 contribution-just-unsure-description = If you come across something that these guidelines don’t cover, please vote according to your best judgement. If you really can’t decide, use the skip button and go on to the next recording.
 
 see-more = <chevron></chevron>See more
-see-less = <chevron></chevron>See less
\ No newline at end of file
+see-less = <chevron></chevron>See less
```
Comment (olejorgenb) [inline web/locales/en/messages.ftl:965]: This snuck into the commit - I leave it since .editorconfig mandates a final newline

File: web/src/components/pages/contribution/speak/speak.tsx [hunk 2]
```diff
@@ -229,26 +229,31 @@ class SpeakPage extends React.Component<Props, State> {
...
     } else if (event.code === 'Digit3' || event.code === 'Numpad3') {
-      index = 2;
+      reRecordIndex = 2;
     } else if (event.code === 'Digit4' || event.code === 'Numpad4') {
-      index = 3;
+      reRecordIndex = 3;
     } else if (event.code === 'Digit5' || event.code === 'Numpad5') {
-      index = 4;
+      reRecordIndex = 4;
+    } else if (event.key === "Backspace" || event.key === "x") {
```
Comment (olejorgenb) [inline web/src/components/pages/contribution/speak/speak.tsx:245]: I used `key` instead of `code` since I think we should respect the user's choice of keyboard layout? I might miss some rationale for using `code` though.

File: web/src/components/pages/contribution/speak/speak.tsx [hunk 3]
```diff
@@ -744,6 +749,13 @@ class SpeakPage extends React.Component<Props, State> {
                 label: 'shortcut-rerecord-toggle-label',
                 action: this.handleRecordClick,
               },
+              {
+                key: 'shortcut-discard-ongoing-recording',
+                label: 'shortcut-discard-ongoing-recording-label',
+                icon: <span>⌫</span>,
```
Comment (olejorgenb) [inline web/src/components/pages/contribution/speak/speak.tsx:755]: A proper icon would be nice, but I'm no artist :) If we can't trust that all fonts have this unicode symbol, a textual representation might be better (but "BACKSPACE" overflows the shortcutbox, so it's will require some work)

File: web/src/services/tracker.ts [hunk 4]
```diff
@@ -69,6 +69,7 @@ export function trackRecording(
     | 'record'
     | 'submit'
     | 'rerecord'
+    | 'discard-ongoing'
```
Comment (olejorgenb) [inline web/src/services/tracker.ts:72]: I think this is sufficient distinct action from "rerecord" to warrant a separate tracking key, but should maybe go in a separate commit?

**Comment by olejorgenb** [comment]
Sidenote: this project have a very nice first-contributor experience. `docker-compose up` and everything works!

**Comment by phirework** [comment]
So glad to hear you had a good with the docker container, that has been a long journey of trial and error 😁 I'm going to tag in @mbransn for a quick design review. Also, I'm thinking this should be ESC instead of / in addition to backspace, thoughts?

**Comment by olejorgenb** [comment]
Yeah, not sure why I didn't think of ESC. It's one of my favorite keys :D

**Comment by mbransn** [comment]
Thanks for this suggestion @olejorgenb :) I agree that having a shortcut to restart recording is a nice addition to the re-record option at the end of the clip. ++ to using `ESC` for the action key and the UI tag.

File: web/src/components/pages/contribution/speak/speak.tsx (hunk 2 above)
Comment (phirework) [reply web/src/components/pages/contribution/speak/speak.tsx:245]: Agreed! Would you be interested in submitting a separate PR to fix the rest of these as well?

File: web/src/components/pages/contribution/speak/speak.tsx (hunk 3 above)
Comment (phirework) [reply web/src/components/pages/contribution/speak/speak.tsx:755]: If we're changing to ESC, I think just using `ESC` would be fine for the "icon"

**Review by phirework — CHANGES_REQUESTED** [review]
As discussed - can you change the trigger key from Backspace to Escape? otherwise this PR is good to go1 Thanks so much :D

File: web/src/components/pages/contribution/speak/speak.tsx (hunk 2 above)
Comment (olejorgenb) [reply web/src/components/pages/contribution/speak/speak.tsx:245]: Its probably easier if you just do it :)

File: web/src/components/pages/contribution/speak/speak.tsx (hunk 3 above)
Comment (olejorgenb) [reply web/src/components/pages/contribution/speak/speak.tsx:755]: Agreed

**Comment by phirework** [comment]
This looks great, thanks!
